src/Models/Concerns/HasConversations.php

```diff
@@ -39,4 +41,19 @@ public function readMessages(): BelongsToMany
         return $this->belongsToMany(config('gossip.models.message'))
             ->withTimestamps();
     }
+
+    /**
+     * Scope a query to only include users of the given conversation.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  int  $conversationId
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeOfConversation(Builder $query, int $conversationId): Builder
+    {
+        return $query->whereHas(
+            'conversations',
+            fn($query) => $query->where('conversations.id', $conversationId),
+        );
```

Comment:
```suggestion
        return $query->where('conversation_id', $conversationId);
```

Reply:
It's a many to many relationship. So this would generate an error (`Unknown column 'conversation_id' in 'where clause'`)